Garage: Add filecoin deal info to rig details page

datadanne opened the pull request:
Adds a button with the filecoin logo on the rig details page that shows filecoin deal information in a modal. 

Is this a good way to present the information @asutula? Or do you have any suggestions on how to make it better?

NOTE: Builds/deploys fail until we've published a new version of the rigs npm-package that includes the new dealsTable in the deployment info
 
![image](https://github.com/tablelandnetwork/rigs/assets/656107/4ddc9376-0c25-4803-83c6-7b30f14ba302)
![image](https://github.com/tablelandnetwork/rigs/assets/656107/72241e08-533d-4f49-bb02-054fb4a020bc)

Closes RIG-3

asutula commented:
Looks great to me. Little nit: "This rigs..." -> "This rig's...". 

I think we'll release the update to the tables and metadata query this Wednesday, so there should be a new rigs package available then.

datadanne commented:
> Looks great to me. Little nit: "This rigs..." -> "This rig's...".
> 
> I think we'll release the update to the tables and metadata query this Wednesday, so there should be a new rigs package available then.

Ah good catch. Fixed!

Sweet, I rebased the branch on top of staging and resolved the conflict. Should be ready to go as soon as the rigs package is available!

sanderpick commented:
very nice. i can do a publish tomorrow while we work through deploying all the filecoin related things.
